infrastructure/simulation-tool/test/operations/changePubKey.test.ts

```diff
@@ -0,0 +1,219 @@
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const expectedParameters: Omit<PreparedPubKeyChange, 'from'> = {
+                ethAuthType: 'ECDSA',
+                feeToken: constants.AddressZero
+            };
+            await executePubKeyChange({ from: senderWallet, ...expectedParameters });
+
+            expect(senderWallet.setSigningKey).to.have.been.calledWith(expectedParameters);
+        });
+    });
+
+    describe('executePubKeyChanges', async () => {
+        it('should call executePubKeyChange for each given preparedPubKeyChange', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+
+            await executePubKeyChanges(preparedPubKeyChanges, 0);
+
+            expect(senderWallet.setSigningKey).to.have.been.calledThrice;
+        });
+
+        it('should call executePubKeyChange with given delay between calls', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+            const delay = 100;
+            const expectedTotalTime = delay * (preparedPubKeyChanges.length - 1);
+
+            const startTime = Date.now();
+            await executePubKeyChanges(preparedPubKeyChanges, delay);
+            const endTime = Date.now();
+
+            const actualTotalTime = endTime - startTime;
+            expect(actualTotalTime).to.be.greaterThan(expectedTotalTime);
+        });
+
+        it('should return array of promises', async () => {
+            const senderWallet = sinon.createStubInstance(RollupWallet);
+            const preparedPubKeyChanges: PreparedPubKeyChange[] = [
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero },
+                { from: senderWallet, ethAuthType: 'ECDSA', feeToken: constants.AddressZero }
+            ];
+
+            const actualResults = await executePubKeyChanges(preparedPubKeyChanges, 0);
+
+            const allResultsAreTransactions = actualResults.reduce((acc, next) => {
+                acc = acc && next instanceof Promise;
+                return acc;
+            }, true);
```

Comment:
Is there any reason for not to use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every)?

Reply:
done